dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx

```diff
@@ -4,15 +4,32 @@ import * as React from "react";
 
 import itBehavesLike from "../../../shared/specs";
 
+import ResourceRef from "shared/ResourceRef";
 import LoadingWrapper from "../../../components/LoadingWrapper";
 import { IIngressSpec, IResource, IServiceSpec, IServiceStatus } from "../../../shared/types";
 import AccessURLItem from "./AccessURLItem";
 import AccessURLTable from "./AccessURLTable";
 
-describe("componentDidMount", () => {
-  it("fetches ingresses", () => {
+const defaultProps = {
+  services: [],
+  ingresses: [],
+  ingressRefs: [],
+  getResource: jest.fn(),
+};
+
+describe("when receiving ingresses", () => {
+  it("fetches ingresses at mount time", () => {
+    const mock = jest.fn();
+    shallow(
+      <AccessURLTable {...defaultProps} ingressRefs={[{} as ResourceRef]} getResource={mock} />,
+    );
+    expect(mock).toHaveBeenCalled();
+  });
+
+  it("fetches new ingresses when received", () => {
     const mock = jest.fn();
-    shallow(<AccessURLTable services={[]} ingresses={[]} fetchIngresses={mock} />);
+    const wrapper = shallow(<AccessURLTable {...defaultProps} getResource={mock} />);
+    wrapper.setProps({ ingressRefs: [{} as ResourceRef] });
     expect(mock).toHaveBeenCalled();
```

Comment:
Ditto here. I realise this line didn't change, but it was previously a mock of the function `fetchIngresses` so it was obvious. now it's a mock of the more generic `getResource`, so it'd be clearer to show that it is indeed getting an Ingress resource.